core/trino-spi/src/main/java/io/trino/spi/connector/Constraint.java

```diff
@@ -134,4 +135,19 @@ public ConnectorExpression getExpression()
     {
         return predicateColumns;
     }
+
+    @Override
+    public String toString()
+    {
+        StringJoiner stringJoiner = new StringJoiner(", ", Constraint.class.getSimpleName() + "[", "]");
+        stringJoiner.add("summary=" + summary);
+        stringJoiner.add("expression=" + expression);
+        if (predicate.isPresent()) {
```

Comment:
Maybe `predicate.ifPresent(p -> stringJoiner.add("predicate=" + p))` and other places too?

Reply:
good catch